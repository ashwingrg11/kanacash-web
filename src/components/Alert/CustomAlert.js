import React from "react";
import { Alert } from "react-bootstrap";
import PropTypes from "prop-types";

const CustomAlert = ({ close, show, variant, message }) => {
  React.useEffect(() => {
    if (show) {
      setTimeout(() => {
        close();
      }, 5000);
    }
  }, [show, close]);

  return (
    <React.Fragment>
      {show && (
        <Alert
          className="custom-alert"
          variant={variant}
          onClose={() => close()}
          dismissible
        >
          <Alert.Heading>{message}</Alert.Heading>
        </Alert>
      )}
    </React.Fragment>
  );
};

CustomAlert.defaultProps = {
  variant: "success",
};

CustomAlert.propTypes = {
  close: PropTypes.func,
  show: PropTypes.bool,
  variant: PropTypes.oneOf(["danger", "success", ""]),
  message: PropTypes.string,
};

export default CustomAlert;
