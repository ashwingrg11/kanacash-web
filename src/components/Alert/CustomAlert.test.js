import React from 'react';
import CustomAlert from './CustomAlert';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('<SendMoney />', () => {
  it('SendMoney renders correctly', () => {
    const tree = renderer
      .create(
          <CustomAlert />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});