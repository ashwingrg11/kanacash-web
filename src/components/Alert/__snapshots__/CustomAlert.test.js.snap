// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendMoney /> SendMoney renders correctly 1`] = `null`;
